event-hubs.api.md

```diff
@@ -0,0 +1,303 @@
+    sendBatch(batch: EventDataBatch, partitionId: string, options?: SendBatchOptions): Promise<void>;
+}
+
+// @public
+export interface EventHubProperties {
+    createdAt: Date;
+    name: string;
+    partitionIds: string[];
+}
+
+// @public
+export class EventPosition {
+    // Warning: (ae-forgotten-export) The symbol "EventPositionOptions" needs to be exported by the entry point index.d.ts
+    // 
+    // @internal
+    constructor(options?: EventPositionOptions);
+    static earliest(): EventPosition;
+    enqueuedTime?: Date | number;
+    static fromEnqueuedTime(enqueuedTime: Date | number): EventPosition;
+    static fromOffset(offset: number, isInclusive?: boolean): EventPosition;
+    static fromSequenceNumber(sequenceNumber: number, isInclusive?: boolean): EventPosition;
+    isInclusive: boolean;
+    static latest(): EventPosition;
+    offset?: number | "@latest";
+    sequenceNumber?: number;
+    }
+
+// @public
+export function extractSpanContextFromEventData(eventData: EventData): SpanContext | undefined;
+
+// @public
+export interface GetPartitionIdsOptions extends AbortSignalOptions, SpanOptions {
+}
+
+// @public
+export interface GetPartitionPropertiesOptions extends AbortSignalOptions, SpanOptions {
+}
+
+// @public
+export interface GetPropertiesOptions extends AbortSignalOptions, SpanOptions {
+}
+
+// @public
+export class InMemoryPartitionManager implements OwnershipManager, CheckpointManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    // (undocumented)
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export interface LastEnqueuedEventInfo {
+    enqueuedTime?: Date;
+    offset?: string;
+    retrievalTime?: Date;
+    sequenceNumber?: number;
+}
+
+export { MessagingError }
+
+// @public
+export interface OwnershipManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+}
+
+// @public
+export interface PartitionCheckpointer {
+    updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
+    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
+    // @internal (undocumented)
+    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
+}
+
+// @public
+export interface PartitionContext {
+    consumerGroupName: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    partitionId: string;
+}
+
+// @public
+export type PartitionManager = CheckpointManager & OwnershipManager;
```

Comment:
Leave as a single interface, rather than splitting it up earlier.

Based on customer feedback we can see if we need to split out or make this more granular.

Reply:
Check with @srnagar before making this change

Reply:
After talking the critical part was not the split of the interfaces - it was making sure the individual contracts for PartitionOwnership and Checkpoint were not intertwined. Changes made and pushed for API review.